### ISSUE TITLE
Setting Different Learning Rate for Backbone and Heads

### ISSUE THREAD

**Issue by jhaggle** [open-mmlab/mmsegmentation#2239]
I am following this tutorial and use my own custom data-set with two classes plus the background class:
https://github.com/open-mmlab/mmsegmentation/blob/master/demo/MMSegmentation_Tutorial.ipynb

And I am reading about “Training tips” in this tutorial:
https://github.com/open-mmlab/mmsegmentation/blob/master/docs/en/tutorials/training_tricks.md#different-learning-ratelr-for-backbone-and-heads

I now want to try to set set different learning rate for backbone and heads. I tried to set it by using the following line in the config file in the notebook tutorial:
`cfg.optimizer = dict(paramwise_cfg = dict(custom_keys={'head': dict(lr_mult=10.)}))`

However, when then running the line 
```
 train_segmentor(model, datasets, cfg, distributed=False, validate=True, 
                meta=dict())
``` 
in the following cell, I get this error:

![bild](https://user-images.githubusercontent.com/11705343/198296521-3a762a42-4e26-46a8-99db-e730333c198d.png)

Is it not possible to set this in the config file?
How do I otherwise set it when running the notebook tutorial?

**Comment by MengzhangLI** [comment]
Hi, it shows types of the `self.base_lr` and `lr_mult` are `NoneType` and `float`, respectively. Please check out your `cfg` again.

**Comment by jhaggle** [comment]
> Hi, it shows types of the `self.base_lr` and `lr_mult` are `NoneType` and `float`, respectively. Please check out your `cfg` again.

Hi, thanks for your answer.
 yes I can see that those variables are NoneType and float. But my questions where above under the image:

Is it not possible to set this in the config file?
How do I otherwise set it when running the notebook tutorial?

**Comment by MeowZheng** [comment]
please use 
`cfg.optimizer.update(dict(paramwise_cfg = dict(custom_keys={'head': dict(lr_mult=10.)})))`

instead of 
`cfg.optimizer = dict(paramwise_cfg = dict(custom_keys={'head': dict(lr_mult=10.)}))`

**Comment by jhaggle** [comment]
> please use `cfg.optimizer.update(dict(paramwise_cfg = dict(custom_keys={'head': dict(lr_mult=10.)})))`
> 
> instead of `cfg.optimizer = dict(paramwise_cfg = dict(custom_keys={'head': dict(lr_mult=10.)}))`

Thank you, that worked! I also tried `cfg.optimizer['paramwise_cfg'] = dict(custom_keys={'head': dict(lr_mult=10.)}))`, which seemed to work as well.
